components/template-mgt/org.wso2.carbon.identity.template.mgt.endpoint/src/main/java/org/wso2/carbon/identity/template/mgt/endpoint/impl/TemplatesApiServiceImpl.java

```diff
@@ -47,98 +47,39 @@
 import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_UNEXPECTED;
 import static org.wso2.carbon.identity.template.mgt.TemplateMgtConstants.ErrorMessages.ERROR_CODE_USER_NOT_AUTHORIZED;
 
-public class
-TemplatesApiServiceImpl extends TemplatesApiService {
+@Deprecated
+public class TemplatesApiServiceImpl extends TemplatesApiService {
 
     private static final Log LOG = LogFactory.getLog(TemplatesApiServiceImpl.class);
 
     @Override
     public Response addTemplate(TemplateDTO template) {
 
-        try {
-            TemplateResponseDTO response = postTemplate(template);
-            return Response.created(getTemplateLocationURI(response.getTemplateName()))
-                    .lastModified(Calendar.getInstance().getTime())
-                    .entity(response)
-                    .build();
-        } catch (TemplateManagementClientException e) {
-            return handleBadRequestResponse(e);
-        } catch (TemplateManagementException e) {
-            return handleServerErrorResponse(e);
-        } catch (Throwable throwable) {
-            return handleUnexpectedServerError(throwable);
-        }
+        return handleForbiddenRequest();
```

Comment:
@GANGANI we are changing the backward compatibility in this deprecating effort. Any particular reasons for this? Can't we only add a depreciate tag?

Reply:
As per the discussion with @emswbandara and the team, we decide to return the forbidden error.